### ISSUE TITLE
java.net.BindException: Can't assign requested address (Bind failed)

### ISSUE THREAD

**Issue by mandrachek** [gretty-gradle-plugin/gretty#191]
I get the following error when attempting to start gretty:

```
Caused by: java.net.BindException: Can't assign requested address (Bind failed)
        at org.akhikhl.gretty.ServiceProtocol.createReader(ServiceProtocol.groovy:20)
```

Looking at line 20 of ServiceProtocol.groovy in gretty-common, it's using InetAddress.localHost, rather InetAddress.loopbackAddress.  On a mac at least, InetAddress.localHost apparently uses the hostname to look up the IP address to bind to.  Unfortunately, I'm on a VPN which is handing out a hostname that currently doesn't resolve to the assigned IP address, so the IP gretty is attempting to bind to isn't bound to any interface on my system. 

Additionally, I don't really want the gretty ports running where I'd have to firewall them.  It seems to me that using InetAddress.loopbackAddress would be more secure and less error prone.

**Comment by boris-petrov** [comment]
@mandrachek - what you say makes sense to me. Mind opening a PR with the change?

**Comment by mandrachek** [comment]
I can try!

**Comment by mikaelvik** [comment]
FYI: this also fixed a `ConnectException` issue for the Windows users on my team, probably related to the `ServerSocket` not being available. 

The issue was resolved when upgrading from `3.0.4` to `3.0.5`. 

Thanks. 

```
Exception in thread "main" java.net.ConnectException: Connection timed out: connect
        at java.net.DualStackPlainSocketImpl.connect0(Native Method)
        at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
        at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.net.Socket.connect(Socket.java:589)
        at java.net.Socket.connect(Socket.java:538)
        at java.net.Socket.<init>(Socket.java:434)
        at java.net.Socket.<init>(Socket.java:244)
        at org.akhikhl.gretty.ServiceProtocol$Writer.write(ServiceProtocol.groovy:76)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```
